src/http.ts

```diff
@@ -39,13 +39,18 @@ function isSubsetErrorCode<T>(code: any, messages: { [c: number]: string }): cod
   return false;
 }
 
-function generateError<E>(errorCode: number, messages: { [c: number]: string }, apiName: string): ResultError<E> {
+function generateError<E>(
+  errorCode: number,
+  messages: { [c: number]: string },
+  apiName: string,
+  error?: string,
+): ResultError<E> {
   if (!isSubsetErrorCode<E>(errorCode, messages)) {
     throw { errorCode, messages, apiName };
   }
 
   const statusCode = errorCode;
-  const statusText = messages[errorCode];
+  const statusText = error ?? messages[errorCode];
```

Comment:
`nit`: it seems weird that we pass in the array of possible messages, and also can pass in an explicit error message now. IMO it would be a lot simpler/clearer if we pass in the explicit error message everywhere we call this (explicit meaning either the "generic" message matching the code or the specific message returned by the API).